src/Filesystem.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Yiisoft\Files;
+
+use League\Flysystem\Filesystem as LeagueFilesystem;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\PathNormalizer;
+use Yiisoft\Aliases\Aliases;
+
+class Filesystem extends LeagueFilesystem implements FilesystemInterface
+{
+    private Aliases $aliases;
+
+    public function __construct(FilesystemAdapter $adapter, array $aliases = [], array $config = [], PathNormalizer $pathNormalizer = null)
+    {
+        if ($aliases !== []) {
+            $aliases = array_merge(['@root' => ''], $aliases);
+            $aliases['@root'] = '';
+        }
+        $this->aliases = new Aliases($aliases);
```

Comment:
Why use aliases with Flysystem if we can eliminate it once and for all, if it is an abstraction root its /.

Reply:
Aliases used not only to shorten the path but also for flexibility. You can use the same alias in your code but the real subpath of the abstract fs may be different.

Reply:
I understand that but if we are going to introduce flysystem, we can use abstraction in everything and eliminate the alias.

Reply:
I think this is not a very good idea to use a new instance of abstract fs instead of every alias.

Reply:
I think that using an abstraction and alias is very bad.

Reply:
> - /public/assets
> - /public/runtime
> - /src/assets
> ```

What happens when you will want to change `runtime` dir path to `/public/temp/runtime`? 
If you use alias `@pub-runtime` => `@root/public/runtime` in your code then you can just change it and any changes of rest code are not necessary.

Reply:
And because changing a directory will have problems, just change the path without using an alias?

Reply:
I really like the availability of the aliases option